src/server/webpack-dev-server.js

```diff
@@ -1,18 +1,18 @@
 import webpack from 'webpack'
 import WebpackDevServer from 'webpack-dev-server'
 
-import config from '../../webpack.config.dev'
+import webpackDevConfig from '../../webpack.config.dev'
 
 
-const WEBPACK_HOST = process.env.WEBPACK_HOST || 'localhost'
-const WEBPACK_PORT = parseInt(process.env.WEBPACK_PORT)
+const WEBPACK_HOST = process.env.WEBPACK_HOST || '0.0.0.0'
```

Comment:
> In this context, what is the difference between `localhost` and `0.0.0.0`?

Reply:
In any context: `localhost` is a hostname that is by default set (in a `hosts` file) to resolve to the [loopback address](https://en.wikipedia.org/wiki/Loopback) `127.0.0.1`.
`0.0.0.0`, on the other hand, is a non-routable meta-address, which means it won't resolve on its own (i.e. you can't access 0.0.0.0 on a client), but is a standard way to express, in a server, “all IPv4 addresses on the local machine”.

Concretely, if server S has 192.168.0.1 as its local address on a local network and client C has 192.168.0.2:

- S can access S resources through 192.168.0.1 and 127.0.0.1.
- C can access S resources through 192.168.0.1.

However, for the server software to listen to 192.168.0.1, S has to explicitly state that it handles incoming packets from 192.168.0.1. In many cases, you'll have a frontal server on S, so you don't need to set this up explicitly: if Nginx listens to 192.168.0.1 and redirects to some local port P, then the S server only needs to bind to 127.0.0.1:P.
Here, we don't have any frontal server. So the only way S can listen both to 192.168.0.1 and 127.0.0.1 (and other addresses from other interfaces) is to use the meta-address 0.0.0.0.

- - - -

In this specific context, this means we're allowing the webpack dev server to be available from all clients, not only the ones on the same computer as the server, so I can do remote debugging from another device on the LAN for example.

Reply:
Thanks a lot for the thorough explanation!